plugin/plugins/filesystem/nodes/ipfs.go

```diff
@@ -0,0 +1,226 @@
+package fsnodes
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/djdv/p9/p9"
+	files "github.com/ipfs/go-ipfs-files"
+	ipld "github.com/ipfs/go-ipld-format"
+	logging "github.com/ipfs/go-log"
+	coreiface "github.com/ipfs/interface-go-ipfs-core"
+	corepath "github.com/ipfs/interface-go-ipfs-core/path"
+)
+
+// IPFS exposes the IPFS API over a p9.File interface
+// Walk does not expect a namespace, only its path argument
+// e.g. `ipfs.Walk([]string("Qm...", "subdir")` not `ipfs.Walk([]string("ipfs", "Qm...", "subdir")`
+type IPFS struct {
+	IPFSBase
+	ipfsHandle
+}
+
+func IPFSAttacher(ctx context.Context, core coreiface.CoreAPI) *IPFS {
+	id := &IPFS{IPFSBase: newIPFSBase(ctx, rootPath("/ipfs"), p9.TypeDir,
+		core, logging.Logger("IPFS"))}
+	id.meta, id.metaMask = defaultRootAttr()
+	return id
+}
+
+func (id *IPFS) Attach() (p9.File, error) {
+	id.Logger.Debugf("Attach")
+	return id, nil
+}
+
+func (id *IPFS) GetAttr(req p9.AttrMask) (p9.QID, p9.AttrMask, p9.Attr, error) {
+	id.Logger.Debugf("GetAttr path: %v", id.Path)
+
+	// For IPFS, we set this up front in Walk
+	return id.Qid, id.metaMask, id.meta, nil
+}
+
+func (id *IPFS) Walk(names []string) ([]p9.QID, p9.File, error) {
+	id.Logger.Debugf("Walk names %v", names)
+	id.Logger.Debugf("Walk myself: %s:%v", id.Path, id.Qid)
+
+	if shouldClone(names) {
+		id.Logger.Debugf("Walk cloned")
+		return []p9.QID{id.Qid}, id, nil
+	}
+
+	var (
+		// returned
+		qids   = make([]p9.QID, 0, len(names))
+		newFid = &IPFS{IPFSBase: newIPFSBase(id.Ctx, id.Path, 0, id.core, id.Logger)}
+		// temporary
+		attr        = &p9.Attr{}
+		requestType = p9.AttrMask{Mode: true}
+		err         error
+		// last-set value is used
+		ipldNode ipld.Node
+	)
+
+	for _, name := range names {
+		callCtx, cancel := context.WithTimeout(id.Ctx, 30*time.Second)
+		defer cancel()
+
+		if newFid.Path, err = id.core.ResolvePath(callCtx, corepath.Join(newFid.Path, name)); err != nil {
```

Comment:
@Stebalien 
If-assignment strictness test 👀
Upon failure this object is discarded. I'm guessing we should separate the lines for safety anyway though.
(I'm going to try to remember to restrict my line-golf in the future)

Reply:
I'd just _always_ avoid using `=` (as opposed to `:=`) in compound if assignment statements.